rfcs/005 - Unsurprising Functions and Scalability RFC.md

```diff
@@ -0,0 +1,215 @@
+
+However, it is possible to fix the SSL cert issue with the current architecture by spinning up a new Fargate cluster running `pgbouncer` with the SSL cert and then pointing the DNS records at that instead of RDS directly, but that would also require the custom routing layer eventually when we need more than one RDS instance. Not more complex than the proposed solution, but not less complex on that front, and with a slightly higher latency as `pgbouncer` won't be running in the same machine as the RDS instance.
+
+If we decided we wanted to keep Graphile Worker and the complexities surrounding it, we could split the scheduler/graphile-worker process from the Express process into two different deployments, making the Express layer autoscaling and the Scheduler layer could be scaled manually with the number of databases managed and eliminate both SPOFs, but the graphile worker cluster would be finicky and likely introduce significant delays in creating new IaSQL databases to manage, at scale needing to pool new database requests together during some batching window and then redeploying the cluster with the new list of databases hardwired into it for each worker process to select one of them to execute. That as well could potentially be improved with a cluster coordination layer like Apache Zookeeper to allow the workers to register which databases they're in charge of and temporarily "double-up" until the cluster is resized, but that introduces much more operational complexity and possible points-of-failure (even if not singular points of failure).
+
+On top of all of that, this direction causes our production and staging environments to become further and further removed from our local and test environments, which increases the maintenance burden and makes the likelihood of bugs getting through higher, as developer testing of certain paths differs more from user usage.
+
+The piecemeal approach to scalability would guide us to a much more complex production environment with worse latencies and a harder-to-manage codebase, and that is why it has been rejected (while we still can).
+
+## Expected Semver Impact
+
+This would be considered a major version bump as the spin-up of new databases will change, as well as how it is deployed in every environment, but if considering only the user experience *after* the database exists, it would just be a patch bump, as it is simply fixing a bug in the RPC functions.
+
+## Affected Components
```

Comment:
So I was debating mentioning that or not, because this repo on its own doesn't deal with any of that. Our production/staging deployment setup does. A bit of where having the RFC inside of the OSS repo breaks down a bit.